core/model/integration.py

```diff
@@ -20,16 +20,37 @@
     from typing_extensions import Self
 
 if TYPE_CHECKING:
-    from core.model import Library
+    from core.model import Collection, Library
 
 
-class IntegrationConfiguration(Base):
+class SettingsModel:
```

Comment:
As I run through the code I see this will not be entirely possible since the integration settings are sometimes accessed directly from the collection object without context of a settings class.
One example is [here](https://github.com/ThePalaceProject/circulation/pull/1189/files#diff-c2b30eb26660a26a6fb7181ed75404502f9c840b3798aedc6dae772c8de17464R361).

Reply:
I've dropped the `setitem` interface and kept the `getitem` interface. Which means there are some places where some shoddy dict.copy() occurs.

Reply:
I'd really like to be able to remove the `getitem` interface here as well, but I can make a follow up ticket to handle that if we don't want to remove it as part of this ticket.

Reply:
This will not remove the need to access the `.settings` attribute in many places throughout the application code because the pydantic models are only known to the API classes.